README.md: Corrected a mistake in the symfony firewall section

galphanet opened the pull request:
The name of the firewall has been corrected from `wsse` to `hmac_auth`

itafroma commented:
Hi, thanks for the contribution! Just to clarify, is this the fix for the problem you experienced in #33 or is this intended to be separate from that?

galphanet commented:
Hi, no it's another one. I'll push the merge request when it will be ready.

galphanet commented:
OK so now it's ready. My proposed changes are for Symfony:
1. Added the ResponseSigner service declaration in documentation as it was never called (seems strange...)
2. base66_encode the secret is not needed in Guzzle init (I send almost 2 days figuring this...)
3. Added HmacClient to use in PHPUnit testing on a Symfony project

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+
+class HmacClient extends Client
```
Can you add a docblock here briefly describing what this is class is for?

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+
+class HmacClient extends Client
+{
+
```
Remove this space

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+     * @param Key $key
+     * @return Key
+     */
+    public function setKey(Key $key) {
```
Opening brace should be on its own line

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+        $this->key = $key;
+
+        return $this;
+
```
Remove extra line

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Sign the request with HMAC HTTP and verify the response signature
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
```
Since this inherits an already-documented method, you can just use:

```php
/**
 * {@inheritdoc}
 */
```

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+     */
+    protected function doRequest($request)
+    {
+
```
Extra line

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+    {
+
+        if(!$this->key instanceof Key) {
+            throw new \Exception("Key not provided");
```
Extending classes should not strengthen preconditions (it's a [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) violation). Instead this should just return a 401 response. For the response message, use single quotation marks and include a period.

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+        $psr7Factory = new DiactorosFactory();
+        $httpFoundationFactory = new HttpFoundationFactory();
+
+        //convert the symfony request to PSR7 request
```
I think we can drop all of these comments: the code is pretty self-documenting.

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+        $authenticator = new ResponseAuthenticator($signedRequest, $this->key);
+
+        if(!$authenticator->isAuthentic($psrResponse)) {
+            throw new \Exception("The response cannot be authenticated !");
```
Same LSP violation here: instead, just return a 401 response. Also, use single quotation marks, and remove the space before the exclamation mark.

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+    protected function doRequest($request)
+    {
+
+        if(!$this->key instanceof Key) {
```
Space between `if` and the opening parenthesis

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+        //init the authenticator and verify if the psr7 response signature is valid
+        $authenticator = new ResponseAuthenticator($signedRequest, $this->key);
+
+        if(!$authenticator->isAuthentic($psrResponse)) {
```
Missing space between `if` and opening parenthesis

itafroma commented on src/Symfony/Tests/HmacClient.php:
```diff
@@ -0,0 +1,78 @@
+        //return the symfony response
+        return $response;
+    }
+
```
Remove extra line

itafroma commented on README.md:
````diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
+    }
+```
+
+
````
Remove extra lines here (only need one).

itafroma commented on README.md:
```diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
+
+        //set the key from the consumer
+        $this->client->setKey(new Key("my-key", "my-not-really-secret"));
+
```
Remove extra line

itafroma commented on README.md:
```diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
+        $this->client = static::createClient();
+
+        //set the key from the consumer
+        $this->client->setKey(new Key("my-key", "my-not-really-secret"));
```
Use single quotation marks for strings

itafroma commented on README.md:
```diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
+    {
+        $this->client = static::createClient();
+
+        //set the key from the consumer
```
I don't think we need this comment: the code is self-documenting.

itafroma commented on README.md:
````diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
 }
 ```
 
+PHPUnit testing a controller behind HMAC HTTP authentification in Symfony :
````
Remove space before the colon.

itafroma commented on README.md:
```diff
@@ -198,6 +203,60 @@ class AppBundle extends Bundle
+PHPUnit testing a controller behind HMAC HTTP authentification in Symfony :
+
+1. Add the service declaration:
+
```
Remove extra line

itafroma commented:
Merged with some changes. Thanks again!
